demos-pipeline/lib/lib/demo-pipeline.ts

```diff
@@ -146,6 +153,56 @@ export class DemoPipelineStack extends cdk.Stack {
         },
       ],
     });
+
+    /** Here we create the logic to check for presence of ECR image on CodeCommit repo creation,
+     * and only start pipeline if image exists. On CodeCommit repo updates, just trigger pipeline.  */
+    const fn = new lambda.Function(this, 'OSImageCheck', {
+      runtime: lambda.Runtime.PYTHON_3_10,
+      handler: 'index.handler',
+      code: lambda.Code.fromInline(`
+import boto3
+
+codecommit_client = boto3.client('codecommit')
+ecr_client = boto3.client('ecr')
+codepipeline_client = boto3.client('codepipeline')
+
+def handler(event, context):
+    response = ecr_client.describe_images(repositoryName='${props.imageRepo.repositoryName}', filter={'tagStatus': 'TAGGED'})
+    for i in response['imageDetails']:
+      if '${props.imageTag}' in i['imageTags']:
+        codepipeline_response = codepipeline_client.start_pipeline_execution(name='${pipeline.pipelineName}')
+      break 
+             `
+      ),
+    });
+
+    const startPipelinePolicy = new iam.PolicyStatement({
+      actions: ['codepipeline:StartPipelineExecution'],
+      resources: [pipeline.pipelineArn],
+    });
+    const ecrPolicy = new iam.PolicyStatement({
+      actions: ['ecr:DescribeImages'],
+      resources: [props.imageRepo.repositoryArn],
+    });
+
+    fn.role?.attachInlinePolicy(
+        new iam.Policy(this, 'CheckOSAndStart', {
+          statements: [startPipelinePolicy, ecrPolicy],
+        }),
+    );
+
+    const ruleOnCreateOrUpdate = new events.Rule(this, 'BuildTriggerRule', {
```

Comment:
can you run `npm run format`? Our CI check is broken...

Reply:
I think we should compare and align the local formatting settings. I might have some old stuff configured in Intelij for js and ts. I will take a look.

Reply:
With the plugins, eslint/prettier should format correctly. VS Code will handle it with the prettier plugin, but JS formatting is excessively complicated.

Reply:
This should be good now. We should add this automatically in npm run build perhaps.